src/dbnode/storage/index_test.go

```diff
@@ -65,6 +71,133 @@ func TestNamespaceIndexCleanupExpiredFilesets(t *testing.T) {
 	require.NoError(t, idx.CleanupExpiredFileSets(now))
 }
 
+func TestNamespaceIndexCleanupDuplicateFilesets(t *testing.T) {
+	md := testNamespaceMetadata(time.Hour, time.Hour*8)
+	nsIdx, err := newNamespaceIndex(md, testShardSet, DefaultTestOptions())
+	require.NoError(t, err)
+
+	idx := nsIdx.(*nsIndex)
+	now := time.Now().Truncate(time.Hour)
+	indexBlockSize := 2 * time.Hour
+	blockTime := now.Add(-2 * indexBlockSize)
+
+	fset1, err := ioutil.TempFile("", "fileset-9000-0-")
+	require.NoError(t, err)
+	fset2, err := ioutil.TempFile("", "fileset-9000-1-")
+	require.NoError(t, err)
+	fset3, err := ioutil.TempFile("", "fileset-9000-2-")
+	require.NoError(t, err)
```

Comment:
Want to create a tempdir and defer remove all to always cleanup even on failure?

```go
	dir, err := ioutil.TempDir("", t.Name()) 
	require.NoError(t, err)

	defer os.RemoveAll(dir)

	fset1, err := ioutil.TempFile(dir, "fileset-9000-0-")
	require.NoError(t, err)
	fset2, err := ioutil.TempFile(dir, "fileset-9000-1-")
	require.NoError(t, err)
	fset3, err := ioutil.TempFile(dir, "fileset-9000-2-")
	require.NoError(t, err)
```